src/main/java/seedu/address/model/property/MatchingPrice.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.model.property;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents matching price of property in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class MatchingPrice {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Matching price should be an integer";
+    public static final String VALIDATION_REGEX = "\\d+";
+    public final String value;
+
+    /**
+     * Constructs a {@code MatchingPrice}.
+     *
+     * @param matchingPrice A valid Matching Price.
+     */
+    public MatchingPrice(String matchingPrice) {
+        requireNonNull(matchingPrice);
+        checkArgument(isValidMatchingPrice(matchingPrice), MESSAGE_CONSTRAINTS);
+        value = matchingPrice;
+    }
+
+    /**
+     * Returns the intersection of ask and bid
+     */
+    public static int getMatchingPrice(Ask ask, Bid bid) {
+        return (Integer.parseInt(ask.toString()) + Integer.parseInt(bid.toString())) / 2;
```

Comment:
maybe we use a getter to the value of the `ask` object instead of converting the string representation to an integer? this is slightly harder to understand